### PR TITLE
add MGE, fix tnfw autodiff

### PR DESCRIPTION
**Pull request by ahuang314**
1. This PR addresses issue #66. In general, whenever `jnp.where(bool, branch_1, branch_2)` is executed, both branches are computed and one is discarded depending on the bool. However, when computing gradients, any NaNs that appear in either branch are propagated forwards, even if that branch is discarded.
2. Implements MGE light profiles for dolphin users

### CODE REVIEW

**Review by sibirrer — APPROVED** [review]
Thank you @ahuang314 !
